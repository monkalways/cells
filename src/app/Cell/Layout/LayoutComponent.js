import React from 'react';
import PropTypes from 'prop-types';
import { Grid, withStyles } from '@material-ui/core';

import constants from '../../constants';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

const Layout = ({ classes, children }) => (
  <Grid container justify="center">
    <Grid container className={classes.container}>
      <Grid item xs={12}>
        {children}
      </Grid>
    </Grid>
  </Grid>
);

Layout.propTypes = propTypes;

export default withStyles((theme) => ({
  container: {
    height: constants.LAYOUT.height,
    [theme.breakpoints.up('lg')]: {
      width: constants.LAYOUT.width,
    },
  },
}))(Layout);
