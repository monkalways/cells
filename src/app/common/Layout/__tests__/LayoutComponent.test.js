import React from 'react';
import { shallow } from 'enzyme';

import { LayoutComponent } from '../LayoutComponent';

describe('Layout', () => {
  let classes;
  let children;

  beforeEach(() => {
    classes = {
      container: 'container',
    };
    children = <h1>Hello Test</h1>;
  });

  const setup = () => shallow(<LayoutComponent classes={classes} children={children} />);

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
