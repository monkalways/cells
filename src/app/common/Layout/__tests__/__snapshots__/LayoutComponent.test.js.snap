// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render 1`] = `
<WithStyles(Grid)
  container={true}
  justify="center"
>
  <WithStyles(Grid)
    className="container"
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <h1>
        Hello Test
      </h1>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
